compiler/AST/symbol.cpp

```diff
@@ -303,6 +308,82 @@ bool VarSymbol::isParameter() const {
   return hasFlag(FLAG_PARAM) || immediate;
 }
 
+
+std::string VarSymbol::docsDirective() {
+  std::string result;
+  if (fDocsTextOnly) {
+    result = "";
+  } else {
+    // TODO: If this is a type (i.e. this->hasFlag(FLAG_TYPE_VARIABLE)), use
+    //       ".. type:: " as directive. (thomasvandoren, 2015-02-21)
+    if (this->isField) {
+      result = ".. attribute:: ";
+    } else {
+      result = ".. data:: ";
+    }
+  }
+  return this->hasFlag(FLAG_CONFIG) ? result + "config " : result;
+}
+
+
+void VarSymbol::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->hasFlag(FLAG_NO_DOC) || this->hasFlag(FLAG_SUPER_CLASS)) {
+      return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+
+  if (this->hasFlag(FLAG_TYPE_VARIABLE)) {
+    *file << "type ";
+  } else if (this->isConstant()) {
+    *file << "const ";
+  } else if (this->isParameter()) {
+    *file << "param ";
+  } else {
+    *file << "var ";
+  }
+
+  *file << this->name;
+
+  if (this->defPoint->exprType != NULL) {
+    *file << ": ";
+    this->defPoint->exprType->prettyPrint(file);
+  }
+
+  // FIXME: Do we want this for non-class globals/configs?
+  //        (thomasvandoren, 2015-02-21)
```

Comment:
Returning to the record/class versus module level argument, to me, the purpose of a type field for a class or record is to give it generic behavior but also to unite certain fields under a single name.  The user of the type chooses what generic instantiation they want, it is apparent that some information must be provided on that account, even if the type designer doesn't want you to access the inner fields themselves.  However, there is no such need to understand the innards of a type alias, you merely need to know what actions you can take on it and where it is required.